app/src/main/java/monster/minions/binocularss/activities/AddFeedActivity.kt

```diff
@@ -154,9 +154,10 @@ class AddFeedActivity : ComponentActivity() {
      * @return url with https:// possibly appended to it.
      */
     private fun addHttps(url: String): String {
+        var trimmedUrl = url.trim(' ');
         return when {
-            url.startsWith("https://") || url.startsWith("http://") -> url
-            else -> "https://$url"
+            trimmedUrl.startsWith("https://") || trimmedUrl.startsWith("http://") -> trimmedUrl
+            else -> "https://$trimmedUrl"
         }
     }
```

Comment:
I would recommend not trimming the URL here, but instead where the input is actually captured in submit() (i.e. the value that is passed into this function). Doing it here kind of violates SOLID as this function should purely be adding http/https, and not trimming the URL. Also, var should not be used here since we are not modifying the variable, instead val should be used to make the variable immutable. Good implementation thought, does this also cover if the user were to somehow insert a newline character ("\n") like Yaniv did?